src/Corporation/Corporation.ts

```diff
@@ -460,13 +461,22 @@ export class Corporation {
     return;
   }
 
+  // Exclude numberOfOfficesAndWarehouses
+  static includedProperties = getKeyList(Corporation, { removedKeys: ["numberOfOfficesAndWarehouses"] });
+
   /** Serialize the current object to a JSON save state. */
   toJSON(): IReviverValue {
-    return Generic_toJSON("Corporation", this);
+    return Generic_toJSON("Corporation", this, Corporation.includedProperties);
   }
 
   /** Initializes a Corporation object from a JSON save state. */
   static fromJSON(value: IReviverValue): Corporation {
+    // numberOfOfficesAndWarehouses is excluded from JSON save state, so we need to calculate it here.
+    value.data.numberOfOfficesAndWarehouses = 0;
+    for (const division of value.data.divisions.values()) {
+      value.data.numberOfOfficesAndWarehouses += getRecordValues(division.offices).length;
+      value.data.numberOfOfficesAndWarehouses += getRecordValues(division.warehouses).length;
+    }
     return Generic_fromJSON(Corporation, value.data);
```

Comment:
Operating on value.data directly isn't great for typesafety since value.data is type `any`. I recommend converting the unknown data into a known class object first.

Whether includedProperties should be passed in to the fromJSON is up for debate, I would probably pass it in but not passing it in shouldn't break anything since that excluded property has never been part of a savegame.

```suggestion
    const corp = Generic_fromJSON(Corporation, value.data, Corporation.includedProperties);
    // numberOfOfficesAndWarehouses is not in the included properties and must be calculated
    for (const division of corp.divisions.values()) {
      corp.numberOfOfficesAndWarehouses += getRecordValues(division.offices).length;
      corp.numberOfOfficesAndWarehouses += getRecordValues(division.warehouses).length;
    }
    return corp;
```

Reply:
I agree with your suggestion and have committed the changes.